components/misc/resourcehelpers.cpp

```diff
@@ -61,30 +61,16 @@ std::string Misc::ResourceHelpers::correctResourcePath(const std::string &topLev
        correctedPath.compare(0, prefix2.size(), prefix2.data()) != 0)
         correctedPath = prefix1 + correctedPath;
 
-    std::string origExt = correctedPath;
-
-    // since we know all (GOTY edition or less) textures end
-    // in .dds, we change the extension
-    bool changedToDds = changeExtensionToDds(correctedPath);
-    if (vfs->exists(correctedPath))
+    if (vfs->exists(correctedPath)
+     || changeExtensionToDds(correctedPath) && vfs->exists(correctedPath))
```

Comment:
You should probably add some parenthesis here, to clarify the grouping:
```
if (vfs->exists(correctedPath)
 || (changeExtensionToDds(correctedPath) && vfs->exists(correctedPath)))
```
Some compilers will actually complain otherwise, since it's not obvious that `&&` binds stronger than `||`.

Reply:
Right. GCC does complain about this.